docs/EditTutorial.md

```diff
@@ -0,0 +1,324 @@
+
+Here is how you could write a book edition view in pure React, leveraging react-admin's [data fetching hooks](./DataProviderIntroduction.md), and [react-hook-form](https://react-hook-form.com/) to bind form inputs with a record object:
+
+```jsx
+import * as React from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { useForm, Controller } from "react-hook-form";
+import { useGetOne, useUpdate } from "react-admin";
+import { Card, TextField, Button, Stack, MenuItem } from "@mui/material";
+
+export const BookEdit = () => {
+  const { id } = useParams();
+  const { handleSubmit, reset, control } = useForm();
+  const { isLoading } = useGetOne(
+    "books",
+    { id },
+    { onSuccess: (data) => reset(data) }
+  );
+  const [update, { isLoading: isSubmitting }] = useUpdate();
+  const navigate = useNavigate();
+  const onSubmit = (data) => {
+    update(
+        "books",
+        { id, data },
+        { onSuccess: () => { navigate('/books'); } }
+    );
+  };
+
+  if (isLoading) return null;
+  return (
+    <div>
+      <Title title="Book Edition" />
+      <Card>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <Stack spacing={2}>
+            <Controller
+              name="title"
+              render={({ field }) => <TextField label="Title" {...field} />}
+              control={control}
+            />
+            <Controller
+              name="author"
+              render={({ field }) => <TextField label="Author" {...field} />}
+              control={control}
+            />
+            <Controller
+              name="availability"
+              render={({ field }) => (
+                <TextField select label="Availability" {...field}>
+                  <MenuItem value="in_stock">In stock</MenuItem>
+                  <MenuItem value="out_of_stock">Out of stock</MenuItem>
+                  <MenuItem value="out_of_print">Out of print</MenuItem>
+                </TextField>
+              )}
+              control={control}
+            />
+            <Button type="submit" disabled={isSubmitting}>
+              Save
+            </Button>
+          </Stack>
+        </form>
+      </Card>
+    </div>
+  );
+};
+```
+
+This form displays 3 inputs (two text inputs and one select input), and redirects to the book list view upon successful submission. It doesn't even contain default values, validation, or dependent inputs. Yet, it's already quite verbose.
+
+It's a super common component. In fact, many of its features could be extracted for reuse in other pages. Let's see how to improve the code and the developer experience in the same process. 
+
+## `<SimpleForm>`: Form Logic In A Simple Layout
+
+React-hook-form requires developers to create a form `control`, and to wrap each input with a `<Controller>` tag. We can put this logic into a custom Form component to avoid repeating it for every form. This is what [the `<SimpleForm>` component](./SimpleForm.md) does:
```

Comment:
It's actually not true. You could just `register` most MUI fields without a controller

Reply:
Did you see this comment?

Reply:
> Most UI libraries are built to support only controlled components, such as [Material-UI](https://github.com/mui-org/material-ui) and [Antd](https://github.com/ant-design/ant-design).

https://react-hook-form.com/advanced-usage#ControlledmixedwithUncontrolledComponents

Is the react-hook-form doc wrong?

Reply:
OK, I'll update our doc then